tool/bindings_generator/generate_bindings.mjs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import * as fs from 'fs';
+import * as child_process from 'child_process';
+import { createRequire } from 'module';
+import * as idl from '@webref/idl';
+import * as assumedTypesModule from './assumedTypes.mjs';
+import * as translatorModule from './translator.mjs';
+import * as util from './util.mjs';
+
+const require = createRequire(import.meta.url);
+const Translator = translatorModule.Translator;
+const assumedTypesBlob = assumedTypesModule.assumedTypesBlob;
+
+function generateHeader() {
+  return (
+`// Copyright (c) ${util.year}, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// TODO(joshualitt): Import js types properly.
+import 'dart:typed_data';
+import 'package:js/js.dart';
+`);
+}
+
+async function translateIDLs() {
+  const files = await idl.listAll();
+  const translator = new Translator();
+  for (const [shortname, file] of Object.entries(files)) {
+    translator.collect(await file.parse());
+  }
+  return translator.translate();
+}
+
+function runDartFormat(filePath) {
+  child_process.exec(`dart format ${filePath}`, (error, stdout, stderr) => {
```

Comment:
Nothing to address now, but I think we can avoid the exec call when this is migrated to Dart and use the dart formatter by source instead.

Reply:
pkg:dart_style, yes!